scripts/checks.sh

```diff
@@ -149,28 +171,35 @@ function check_networking()
 }
 
 function check_under_voltage(){
-	if dmesg | grep -q "Under-voltage detected\!"; then
-		log_status "${BAD}" "${FUNCNAME[0]}" "Under-voltage events detected, check/change the power supply ASAP"
-	else
-		log_status "${GOOD}" "${FUNCNAME[0]}" "No under-voltage events detected"
+	local SLUG_WHITELIST=('raspberrypi3-64' 'raspberrypi4-64' 'raspberry-pi' 'raspberry-pi2' 'raspberrypi3' 'fincm3')
+	if is_valid_check WHITELIST "${SLUG_WHITELIST[*]}"; then
+		if dmesg | grep -q "Under-voltage detected\!"; then
+			log_status "${BAD}" "${FUNCNAME[0]}" "Under-voltage events detected, check/change the power supply ASAP"
+		else
+			log_status "${GOOD}" "${FUNCNAME[0]}" "No under-voltage events detected"
+		fi
 	fi
 }
 
 function check_temperature(){
-	local -i temp
-	local -i therm_count=0
-	for i in /sys/class/thermal/thermal* ; do
-		if [ -e "$i/temp" ]; then
-			therm_count+=1
-			temp=$(cat "$i/temp")
-			if (( temp >= 80000 )); then
-				log_status "${BAD}" "${FUNCNAME[0]}" "Temperature above 80C detected ($i)"
-				return
+	# see https://github.com/balena-io/device-diagnostics/issues/168
+	local SLUG_BLACKLIST=('jetson-nano' 'jn30b-nano')
+	if is_valid_check BLACKLIST "${SLUG_BLACKLIST[*]}"; then
```

Comment:
This is ok. but I'm wondering if we are polluting the script with some semantics and logic that should live elsewhere..

Reply:
That's a fair point, especially given that this bloat must be sent over the network to execute. I'm open to other implementation ideas. I had hacked up a templated version of this script for the proxy to inject values (specifically to deduplicate the `$ENG` var snippet). Maybe there's a cleaner way server-side to cobble this together (though that also may be the wrong place)?

cc @wrboyce for curiosity